lib/ansible/modules/cloud/docker/docker_image.py

```diff
@@ -788,15 +788,41 @@ def load_image(self):
         '''
         # Load image(s) from file
         load_output = []
+        has_output = False
         try:
             self.log("Opening image %s" % self.load_path)
             with open(self.load_path, 'rb') as image_tar:
                 self.log("Loading image from %s" % self.load_path)
-                for line in self.client.load_image(image_tar):
-                    self.log(line, pretty_print=True)
-                    if "stream" in line or "status" in line:
-                        load_line = line.get("stream") or line.get("status") or ''
-                        load_output.append(load_line)
+                output = self.client.load_image(image_tar)
+                if output is not None:
+                    # Old versions of Docker SDK of Python (before version 2.5.0) do not return anything.
+                    # (See https://github.com/docker/docker-py/commit/7139e2d8f1ea82340417add02090bfaf7794f159)
+                    # Note that before that commit, something else than None was returned, but that was also
+                    # only introduced in a commit that first appeared in 2.5.0 (see
+                    # https://github.com/docker/docker-py/commit/9e793806ff79559c3bc591d8c52a3bbe3cdb7350).
+                    # So the above check works for every released version of Docker SDK for Python.
+                    has_output = True
+                    for line in output:
+                        self.log(line, pretty_print=True)
+                        if "stream" in line or "status" in line:
+                            load_line = line.get("stream") or line.get("status") or ''
+                            load_output.append(load_line)
+                else:
+                    if LooseVersion(docker_version) < LooseVersion('2.5.0'):
+                        self.client.module.warn(
+                            'The installed version of the Docker SDK for Python does not return the loading results'
+                            ' from the Docker daemon. Therefore, we cannot verify whether the expected image was'
+                            ' loaded, whether multiple images where loaded, or whether the load actually succeeded.'
+                            ' If you are not stuck with Python 2.6, *please* upgrade to a version newer than 2.5.0'
+                            ' (2.5.0 was released in August 2017).'
+                        )
+                    else:
+                        self.client.module.warn(
+                            'The API version of your Docker daemon is < 1.23, which does not return the image'
```

Comment:
@felixfontein  Do we know for sure this is the only other way to get `None` above? Would it be better to check `self.client.version()` and ensure this is true before we claim it?

Reply:
Let's say it like this: according to Docker's API documentation, this is the only way this could/should happen. It could be that this results in the wrong message if Docker implemented something else than they claim, or in case the user uses another API (like Podman's docker API - I have no idea if that still is a thing, and if it is, whether it suports this API endpoint at all, and if it does, what it returns). I think it's better to return this message in case `None` is returned instead of crashing, or having another branch for the (hopefully) unlikely case that Docker did something wrong or that a more or less "compatible" API is used that turns out not to be compatible enough.

Reply:
@felixfontein sounds good to me, just wanted to check. :-)